Add examples for new explosion bot commands (#11082)
explosion/spaCy: extra/DEVELOPER_DOCS/ExplosionBot.md

Some things to note:

- The `@explosion-bot please` must be the beginning of the command - you cannot add anything in front of this or else the robot won't know how to parse it. Adding anything at the end aside from the test name will also confuse the robot, so keep it simple!
- The command name (such as `test_gpu`) must be one of the tests that the bot knows how to run. The available commands are documented in the bot's [workflow config](https://github.com/explosion/spaCy/blob/master/.github/workflows/explosionbot.yml#L26) and must match exactly one of the commands listed there.
- The robot can't do multiple things at once, so if you want it to run multiple tests, you'll have to summon it with one comment per test.

### Examples

- Execute spaCy slow GPU tests with a custom thinc branch from a spaCy PR:

  ```
  @explosion-bot please test_slow_gpu --thinc-branch <branch_name>
  ```

  `branch_name` can either be a named branch, e.g: `develop`, or an unmerged PR, e.g: `refs/pull/<pr_number>/head`.

- Execute spaCy Transformers GPU tests from a spaCy PR:

  ```
  @explosion-bot please test_gpu --run-on spacy-transformers --run-on-branch master --spacy-branch current_pr
  ```

>   This will launch the GPU pipeline for the `spacy-transformers` repo, use the current PR's branch to build spaCy, and execute the tests in the `master` branch of the former.

Reviewer: ```suggestion
  This will launch the GPU pipeline for the `spacy-transformers` repo on its `master` branch, using the current PR's branch to build spaCy.
```
